frontend/catalyst/jax_tracer.py

```diff
@@ -522,6 +528,9 @@ def trace_observables(
         coeffs = jax.numpy.array(terms[0])
         nested_obs = trace_observables(terms[1][0], qrp, m_wires)[0]
         obs_tracers = hamiltonian_p.bind(coeffs, nested_obs)
+    elif paulis := obs._pauli_rep:  # pylint: disable=protected-access
```

Comment:
Move pauli rep at the end because op that were previously hamiltonian are now sum